packages/flutter_tools/lib/src/device.dart

```diff
@@ -278,6 +278,28 @@ abstract class DeviceManager {
       return DeviceDiscoverySupportFilter.excludeDevicesUnsupportedByFlutter();
     }
   }
+
+  /// If the user did not specify to run all or a specific device, then attempt
+  /// to prioritize ephemeral devices.
+  ///
+  /// If there is not exactly one ephermeral device return all devices.
+  ///
+  /// For example, if the user only typed 'flutter run' and both an Android
+  /// device and desktop device are available, choose the Android device.
+  ///
+  /// Note: ephemeral is nullable for device types where this is not well
+  /// defined.
+  List<Device> prioritizeEphemeralDevices(List<Device> devices){
```

Comment:
This method name sounds like it should return a bool, like whether it's prioritized, or that it returns a list of the devices with the ephemeral devices sorted first. 
How about `filterToSingleEphemeralDeviceIfNeeded` (or something better than that, it's clunky)?

Reply:
What do you think of `getSoleEphemeralDevice` and change return type to `Device?` and return null if not found.

Reply:
I like the `Device?` pattern better.  How about `getSingleEphemeralDevice` instead of "sole" since "single" has a meaning in dart (though we def don't need to throw the StateError: https://api.dart.dev/be/180791/dart-core/Iterable/single.html)